wandb/sdk/wandb_settings.py

```diff
@@ -1395,7 +1395,11 @@ def _infer_settings_from_environment(
                 # chroot jails or docker containers. Return user id in these cases.
                 settings["username"] = str(os.getuid())
 
-        settings["_executable"] = sys.executable
+        # one special case here is running inside a PEX environment,
+        # see https://pex.readthedocs.io/en/latest/index.html for more info about PEX
+        settings["_executable"] = (
+            self._executable or os.environ.get("PEX") or sys.executable
```

Comment:
I was just looking at the [official docs](https://docs.python.org/3/library/sys.html#sys.executable) and sys.executable can return empty string or None.  I imagine it makes more sense to fall back to `"python"` as the executable if we can't get an absolute path...